streams/src/main/java/org/apache/kafka/streams/kstream/Window.java

```diff
@@ -17,20 +17,37 @@
 
 package org.apache.kafka.streams.kstream;
 
-import org.apache.kafka.streams.processor.ProcessorContext;
-import org.apache.kafka.streams.processor.StateStore;
-
-import java.util.Iterator;
-
-public interface Window<K, V> extends StateStore {
-
-    void init(ProcessorContext context);
-
-    Iterator<V> find(K key, long timestamp);
-
-    Iterator<V> findAfter(K key, long timestamp);
-
-    Iterator<V> findBefore(K key, long timestamp);
-
-    void put(K key, V value, long timestamp);
+public class Window implements Comparable<Window> {
+
+    private long start;
+    private long end;
+
+    public Window(long start, long end) {
+        this.start = start;
+        this.end = end;
+    }
+
+    /**
+     * Returns the start timestamp of this window, inclusive
+     */
+    public long start() {
+        return start;
+    }
+
+    /**
+     * Returns the end timestamp of this window, exclusive
+     */
+    public long end() {
+        return end;
+    }
+
+    @Override
+    public int compareTo(Window other) {
+        if (this.start() > other.start())
```

Comment:
`Window`s are compared only by starts? Are they equal even when ends differ?
btw, if we compare both starts and ends, `Window` no longer has a total ordering, thus no longer `Comparable`...